pkg/terraform/terraform.go

```diff
@@ -31,6 +31,9 @@ func newTFExec(datadir string) (*tfexec.Terraform, error) {
 	tf.SetStderr(lpError)
 	tf.SetLogger(newPrintfer())
 
+	// Opt in to the beta of v3.0 of the azurerm terraform provider.
+	tf.SetEnv(map[string]string{"ARM_THREEPOINTZERO_BETA": "true"})
```

Comment:
I think that this may be preventing environment variables from the installer context from getting passed to the terraform context, blocking the usage of environment variables set for AWS credentials.